packages/components/src/chart/d3chart/test/utils.js

```diff
@@ -157,6 +158,228 @@ describe( 'getXLineScale', () => {
 	} );
 } );
 
+describe( 'getXTicks', () => {
```

Comment:
These tests. 💯 

Thanks so much for taking the time to add some coverage around this logic.